pilot/pkg/xds/endpoint_builder.go

```diff
@@ -78,9 +83,53 @@ func (b EndpointBuilder) Key() string {
 	if b.service != nil {
 		params = append(params, string(b.service.Hostname)+"/"+b.service.Attributes.Namespace)
 	}
+	params = append(params, b.multinetworkKey()...)
 	return strings.Join(params, "~")
 }
 
+func (b EndpointBuilder) multinetworkKey() []string {
```

Comment:
I'm concerned that the EDS cache issues with multicluster/multinetwork. 

- When new endpoints for a service that's already in the cache are discovered in a different cluster-registry, will the cache be evicted? Does this apply just to multicluster? 
- For multi-network it's clear that as-is, changes to the multi-network config will not effect EDS and we'll just be using cached responses. This PR tries to fix some of that but needs a lot of cleanup.

Reply:
Thoughts on making multi-network "uncacheable" in this PR then implementing that properly in a follow-up? @howardjohn